File: src/main/java/com/movisens/xs/api/models/ApiError.java
package com.movisens.xs.api.models;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

public class ApiError {
    public String error;
    public String error_description;

    public ApiError(String error, String error_description) {
        this.error = error;
        this.error_description = error_description;
    }

    public String getMessage() {
        return error + ": " + error_description;
    }

    public void setMessage(String error_description) {
        this.error_description = error_description;
    }

    public static ApiError parse(String str) {
        Gson gson = new GsonBuilder().create();
        ApiError apiError = gson.fromJson(str, ApiError.class);
        return apiError;
    }

    @Override
    public String toString() {
        return "ApiError{" +
                "error='" + error + '\'' +
                ", error_description='" + error_description + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ApiError apiError = (ApiError) o;

        if (error != null ? !error.equals(apiError.error) : apiError.error != null) return false;
        return error_description != null ? error_description.equals(apiError.error_description) : apiError.error_description == null;
    }

    @Override
    public int hashCode() {
        int result = error != null ? error.hashCode() : 0;
        result = 31 * result + (error_description != null ? error_description.hashCode() : 0);
        return result;
    }
}

File: src/main/java/com/movisens/xs/api/models/ApiError.kt
package com.movisens.xs.api.models

import com.google.gson.GsonBuilder

class ApiError(var error: String, var error_description: String) {
    var message: String
        get() = "$error: $error_description"
        set(error_description) {
            this.error_description = error_description
        }

    override fun toString(): String {
        return "ApiError{" +
                "error='" + error + '\'' +
                ", error_description='" + error_description + '\'' +
                '}'
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val apiError = other as ApiError
        if (error != apiError.error) return false
        return error_description == apiError.error_description
    }

    override fun hashCode(): Int {
        var result = error.hashCode()
        result = 31 * result + error_description.hashCode()
        return result
    }

    companion object {
        @JvmStatic
        fun parse(str: String?): ApiError {
            val gson = GsonBuilder().create()
            return gson.fromJson(str, ApiError::class.java)
        }
    }
}